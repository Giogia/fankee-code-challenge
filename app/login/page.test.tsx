import { render } from '@testing-library/react'

jest.mock('react-dom', () => ({
   ...jest.requireActual('react-dom'),
   useFormState: jest.fn().mockReturnValue([{ message: '' }, jest.fn()]),
   useFormStatus: jest.fn().mockReturnValue({ pending: false, action: jest.fn() })
}))

import Login from './page'

describe('Login', () => {
   it('renders correctly', () => {
      const { asFragment } = render(<Login />)
      expect(asFragment()).toMatchSnapshot()
   })
})