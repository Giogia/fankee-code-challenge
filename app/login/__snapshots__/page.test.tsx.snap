// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<DocumentFragment>
  <main
    class="flex min-h-screen flex-col items-center gap-5 justify-center p-8 animate-in-up"
  >
    <a
      href="/"
    >
      <img
        alt="Fankee Logo"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="40"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
    </a>
    <form
      class="w-full max-w-2xl"
    >
      <div
        class="flex flex-col gap-8 p-6 bg-gray-200 rounded-lg shadow-md animate-in-up"
        data-testid="card"
      >
        <div
          class="header"
        >
          <div
            class="flex flex-col gap-2"
          >
            <h1
              class="text-4xl font-extrabold text-gray-800 animate-in"
            >
              Stay Tuned
            </h1>
            <div>
              <p
                class="text-md text-gray-800 animate-in"
              >
                Be part of something big!
              </p>
              <p
                class="text-md text-gray-800 animate-in"
              >
                Stay tuned to be the first to know about our upcoming launch!
              </p>
            </div>
          </div>
        </div>
        <div
          class="body"
        >
          <div
            class="flex flex-col gap-2"
          >
            <input
              class="inline-flex w-full border-2 border-gray-300 focus:backdrop-blur-sm text-sm text-gray-400 bg-transparent h-10 px-3 rounded-lg transition duration-250 ease-in-out animate-in-up"
              id="email"
              name="email"
              placeholder="Insert your email"
              type="email"
              value=""
            />
            <p
              class="text-xs text-gray-800 animate-in"
            >
              You will receive an email containing a "Magic Link" that can be used to sign in without the need for a password.
            </p>
          </div>
        </div>
        <div
          class="actions"
        >
          <div
            class="flex justify-end"
          >
            <button
              class="rounded-full inline-flex gap-2 px-5 py-2.5 cursor-pointer hover:text-gray-400 transition duration-300 ease-in-out active:scale-95 animate-in-up bg-yellow-400 hover:bg-yellow-300"
              type="submit"
            >
              <p
                class="text-xs text-gray-800 animate-in"
              >
                Send Email
              </p>
            </button>
          </div>
        </div>
      </div>
    </form>
  </main>
</DocumentFragment>
`;
